### PR TITLE
modify makehosts_n_r testcase for makehosts timing issue

### PR DESCRIPTION
**Pull request by junxiawang**
Modify makehosts_n_r  testcase for makehost timing issue.
Before the modify after makehosts 30s ,makehosts could modify "/etc/hosts".So I add sleep loop here for makehosts timing issue.

### CODE REVIEW

File: xCAT-test/autotest/testcase/makehosts/cases0 [hunk 1]
```diff
@@ -134,43 +134,37 @@ cmd:lsdef s01r1b01;if [ $? -eq 0 ]; then lsdef -l s01r1b01 -z >/tmp/s01r1b01.sta
...
 cmd:nodeadd s01 groups=service; chdef s01 ip=70.2.0.254;nodeadd s01r1b01 groups=compute; chdef s01r1b01 ip=80.2.0.254;makehosts
 check:rc==0
-cmd:sleep 20
-cmd:rc1=`cat /etc/hosts | grep "70.2.0.254" | grep s01`;rc2=`cat /etc/hosts | grep "80.2.0.254" | grep s01r1b01`;if [[ $rc1 =~ "70.2.0.254 s01" ]] && [[ $rc2 =~ "80.2.0.254 s01r1b01" ]];then exit 0;else exit 1;fi
+cmd:a=0;while true; do [ $a -eq 60 ] && exit 1;rc=`cat /etc/hosts`;if [[ $rc =~ "70.2.0.254 s01" ]] && [[ $rc =~ "80.2.0.254 s01r1b01" ]];then exit 0;else a=$[$a+1];sleep 1;echo $a;fi;done
 check:rc==0
 cmd:makehosts s01
 check:rc==0
-cmd:sleep 20
 cmd:cat /etc/hosts
```
Comment (hu-weihua) [inline xCAT-test/autotest/testcase/makehosts/cases0]: Suggest line 141 change to 

     cmd: grep "s01" /etc/hosts

**Comment by bybai** [comment]
I am OK to merge.

**Comment by hu-weihua** [comment]
agree to merge.

**Comment by daniceexi** [comment]
I don't understand what the change was from the description words ``Before the modify after makehosts 30s ,makehosts could modify "/etc/hosts".`` Please refine it.

**Comment by bybai** [comment]
From the auto test result, after the test case is added /etc/hosts check, all the test cases are passed.

File: xCAT-test/autotest/testcase/makehosts/cases0 [hunk 2]
```diff
@@ -134,43 +134,37 @@ cmd:lsdef s01r1b01;if [ $? -eq 0 ]; then lsdef -l s01r1b01 -z >/tmp/s01r1b01.sta
 check:rc==0
 cmd:nodeadd s01 groups=service; chdef s01 ip=70.2.0.254;nodeadd s01r1b01 groups=compute; chdef s01r1b01 ip=80.2.0.254;makehosts
 check:rc==0
-cmd:sleep 20
-cmd:rc1=`cat /etc/hosts | grep "70.2.0.254" | grep s01`;rc2=`cat /etc/hosts | grep "80.2.0.254" | grep s01r1b01`;if [[ $rc1 =~ "70.2.0.254 s01" ]] && [[ $rc2 =~ "80.2.0.254 s01r1b01" ]];then exit 0;else exit 1;fi
+cmd:a=0;while true; do [ $a -eq 60 ] && exit 1;rc=`cat /etc/hosts`;if [[ $rc =~ "70.2.0.254 s01" ]] && [[ $rc =~ "80.2.0.254 s01r1b01" ]];then exit 0;else a=$[$a+1];sleep 1;echo $a;fi;done
```
Comment (daniceexi) [inline xCAT-test/autotest/testcase/makehosts/cases0]: I think we need to add comment for why the this change fixes the problem.

**Pull request merged**